lib/dry/types/hash/schema.rb

```diff
@@ -57,8 +57,18 @@ def try(hash, &block)
           end
         end
 
+        def to_ast
+          [:hash,
+           [hash_type,
+            [:member_types, [member_types.values.map(&:to_ast)]]]]
```

Comment:
We also need names, so `member_types.map { |name, type| [:member, [name, type.to_ast]] }` should be OK :)